src/navigation-bar.tsx

```diff
@@ -98,27 +105,133 @@ type NavigationBarContentContainerProps = {
     right?: React.ReactNode;
     children?: React.ReactNode;
     desktopOnly?: boolean;
+    expandRightContent?: boolean;
```

Comment:
I needed this because in desktop MainNavigationBar, the navbar sections take up the remaining space, but in other cases the right content does that (and this auxiliary component is reused by all navbars)